madmom/ml/hmm.pyx

```diff
@@ -394,6 +394,38 @@ class HiddenMarkovModel(object):
             raise ValueError('Initial distribution is not a probability '
                              'distribution.')
         self.initial_distribution = initial_distribution
+        # attributes needed for stateful processing (i.e. forward_step())
+        self._prev = self.initial_distribution.copy()
+        self._cur = np.zeros_like(self._prev)
```

Comment:
Ok, I opted against calling `reset()`, because I think it is more explicit to initialise variables in `__init__()` and not in any other method.